### DISCUSSION TITLE
Unable to get accurate bounding box for table structure recognition

### DISCUSSION THREAD

**Discussion by Bishow-99** [PaddlePaddle/PaddleOCR#13499] [Q&A]
I have trained the table structure Recognition model on bank statements which has a borderless table, like

![349471132-ddf81eb2-f3ff-4139-9642-0a1dd522856d](https://github.com/user-attachments/assets/c9c96d00-0be4-4f22-b993-948dd8164029)

For above image I have a annotation file as
{"filename": "bank_statement.png", "html": {"structure": {"tokens": ["", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", ""]}, "cells": [{"tokens": ["D", "a", "t", "e"], "bbox": [[10, 1], [84, 1], [84, 22], [10, 22]]}, {"tokens": ["D", "e", "s", "c", "r", "i", "p", "t", "i", "o", "n"], "bbox": [[111, 1], [723, 1], [723, 22], [111, 22]]}, {"tokens": ["A", "m", "o", "u", "n", "t"], "bbox": [[952, 1], [1014, 1], [1014, 20], [952, 20]]}, {"tokens": ["1", "1", "/", "0", "2", "/", "2", "2"], "bbox": [[11, 28], [83, 28], [83, 72], [11, 72]]}, {"tokens": ["S", "q", "u", "a", "r", "e", " "," "," ", "I", "n", "c", "D", "E", "S", ":", "2", "2", "1", "1", "0", "2", "P", "2", " ", "I", "D", ":", "L", "2", "0", "8", "7", "6", "1", "2", "4", "3", "7", "7", "7", " ", "I", "N", "D", "N", ":", "B", "u", "n", "a", " ", "E", "n", "t", "e", "r", "p", "r", "i", "s", "e", " ", "I", "n", "c", " ", "D", "B", " ", "C", "O", "I", "D", ":", "9", "4", "2", "4", "3", "0", "0", "0", "0", "2", " ", "C", "C", "D"], "bbox": [[109, 29], [725, 29], [725, 73], [109, 73]]}, {"tokens": ["2", "9", "9", ".", "7", "6"], "bbox": [[952, 29], [1019, 29], [1019, 75], [952, 75]]}, {"tokens": ["1", "1", "/", "0", "3", "/", "2", "2"], "bbox": [[10, 82], [82, 82], [82, 123], [10, 123]]}, {"tokens": ["S", "q", "u", "a", "r", "e", " ", "I", "n", "c"," "," "," ", "D", "E", "S", ":", "2", "2", "1", "1", "O", "3", "P", "2", " ", "I", "D", ":", "L", "2", "0", "8", "7", "6", "1", "5", "0", "2", "1", "1", "1", " ", "I", "N", "D", "N", ":", "B", "u", "n", "a", " ", "E", "n", "t", "e", "r", "p", "r", "i", "s", "e", " ", "I", "n", "c", " ", "D", "B", " ", "C", "C", "I", "D", ":", "9", "4", "2", "4", "3", "0", "0", "0", "0", "2", " ", "C", "C", "D"], "bbox": [[110, 82], [725, 82], [725, 130], [110, 130]]}, {"tokens": ["5", "5", "7", ".", "7", "2"], "bbox": [[959, 82], [1023, 82], [1023, 129], [959, 129]]}, {"tokens": ["1", "1", "/", "0", "4", "/", "2", "2"], "bbox": [[9, 136], [82, 136], [82, 168], [9, 168]]}, {"tokens": ["S", "q", "u", "a", "r", "e", " ", "I", "n", "c"," "," "," ", "D", "E", "S", ":", "2", "2", "1", "1", "0", "4", "P", "2", "I", "D", ":", "L", "2", "0", "8", "7", "6", "1", "8", "3", "3", "4", "5", "1", "I", "N", "D", "N", ":", "B", "u", "n", "a", " ", "E", "n", "t", "e", "r", "p", "r", "i", "s", "e", " ", "I", "n", "c", " ", "D", "B", "C", "O", "I", "D", ":", "9", "4", "2", "4", "3", "0", "0", "0", "0", "2", " ", "C", "C", "D"], "bbox": [[109, 136], [727, 136], [727, 177], [109, 177]]}, {"tokens": ["5", "2", "6", ".", "1", "6"], "bbox": [[960, 137], [1023, 137], [1023, 175], [960, 175]]}]}, "gt":"<html><body><table><thead><tr><td>Date</td><td>Description</td><td>Amount</td></tr></thead><tbody><tr><td>11/02/22</td><td>Square   IncDES:221102P2 ID:L208761243777 INDN:Buna Enterprise Inc DB COID:9424300002 CCD</td><td>299.76</td></tr><tr><td>11/03/22</td><td>Square Inc   DES:2211O3P2 ID:L208761502111 INDN:Buna Enterprise Inc DB CCID:9424300002 CCD</td><td>557.72</td></tr><tr><td>11/04/22</td><td>Square Inc   DES:221104P2ID:L208761833451INDN:Buna Enterprise Inc DBCOID:9424300002 CCD</td><td>526.16</td></tr></tbody></table></body></html>"}

I drew the annotated bounding box and it looks as

![349471685-141e3d25-f181-4bb5-88fe-71b67e25bbfe](https://github.com/user-attachments/assets/0f29a489-d7f4-4451-b623-6000cc357761)

But after training although I am getting right HTML structure but getting inaccurate coordinates result for each HTML structure even for the same image that I have used while training and looks as

![349472267-04cdf039-b760-4974-bb62-04868d0e3437](https://github.com/user-attachments/assets/000225b1-8d05-4ef6-a60e-9524af944a50)

Please help me to figure out why I am getting in-accurate bounding box even though the HTML structure is getting 100% accurate, Thank you so much in advance

**Comment by SWHL** [comment]
related with [Discussion](https://github.com/PaddlePaddle/PaddleOCR/discussions/13509)
